src/include/server/mir/shell/accessibility_manager.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_SHELL_ACCESSIBILITY_MANAGER_H
+#define MIR_SHELL_ACCESSIBILITY_MANAGER_H
+
+#include <memory>
+#include <vector>
+
+namespace mir
+{
+namespace frontend
+{
+class KeyboardCallbacks;
+class KeyboardHelper;
+}
+namespace shell
+{
+class AccessibilityManager
+{
+public:
+    void register_keyboard_helper(std::shared_ptr<frontend::KeyboardHelper>);
```

Comment:
Looks wrong: the alternative is to declare a `shell::KeyboardHelper` interface (which `frontend::KeyboardHelper` could implement)